docs/static/config-details.asciidoc

```diff
@@ -76,6 +76,15 @@ process.
 info, see <<profiling-the-heap>>.
 // end::heap-size-tips[]
 
+
+[[off-heap-size]]
+==== Setting the off-heap size
+
+Besides heap, others like OS, persistent queue mmap pages and direct memory need memory in addition of heap memory.
+By default, a JVM's off-heap direct memory limit is the same as the heap size, see <<plugins-inputs-beats#plugins-inputs-beats-memory, beats input memory usage>>.
+Consider setting `-XX:MaxDirectMemorySize` to half of the heap size.
```

Comment:
```suggestion
The operating system, persistent queue mmap pages, direct memory, and other processes require memory in addition to memory allocated to heap size.
Keep the overall memory requirements in mind when you allocate memory.

By default, a JVM's off-heap direct memory limit is the same as the heap size, check out <<plugins-inputs-beats-memory,beats input memory usage>>.
Consider setting `-XX:MaxDirectMemorySize` to half of the heap size.```

Reply:
Suggested rewording for discussion and iteration. Also fixed the link format to pass docs-ci.